examples/structured_2d_dgsem/elixir_euler_convergence_implicit_sparse_jacobian.jl

```diff
@@ -0,0 +1,105 @@
+# sparsity-detection ready datatype, which is retrieved here
+jac_eltype = jacobian_eltype(real(solver), jac_detector)
+
+# Semidiscretization for sparsity pattern detection
+semi_jac_type = SemidiscretizationHyperbolic(mesh, equations,
+                                             initial_condition_convergence_test,
+                                             solver,
+                                             source_terms = source_terms_convergence_test,
+                                             uEltype = jac_eltype) # Need to supply Jacobian element type
+
+tspan = (0.0, 5.0) # Re-used for wrapping `rhs` below
+
+# Call `semidiscretize` to create the ODE problem to have access to the
+# initial condition based on which the sparsity pattern is computed
+ode_jac_type = semidiscretize(semi_jac_type, tspan)
+u0_ode = ode_jac_type.u0
+du_ode = similar(u0_ode)
+
+###############################################################################
+### Compute the Jacobian sparsity pattern ###
+
+# Wrap the `Trixi.rhs!` function to match the signature `f!(du, u)`, see
+# https://adrianhill.de/SparseConnectivityTracer.jl/stable/user/api/#ADTypes.jacobian_sparsity
+rhs_wrapped! = (du_ode, u0_ode) -> Trixi.rhs!(du_ode, u0_ode, semi_jac_type, tspan[1])
+
+jac_prototype = jacobian_sparsity(rhs_wrapped!, du_ode, u0_ode, jac_detector)
+
+# For most efficient solving we also want the coloring vector
+
+coloring_prob = ColoringProblem(; structure = :nonsymmetric, partition = :column)
+coloring_alg = GreedyColoringAlgorithm(; decompression = :direct)
+coloring_result = coloring(jac_prototype, coloring_prob, coloring_alg)
+coloring_vec = column_colors(coloring_result)
+
+###############################################################################
+### sparsity-aware semidiscretization and ode ###
+
+# Semidiscretization for actual simulation
+semi_float_type = SemidiscretizationHyperbolic(mesh, equations,
+                                               initial_condition_convergence_test,
+                                               solver,
+                                               source_terms = source_terms_convergence_test)
+
+# Supply Jacobian prototype and coloring vector to the semidiscretization
+ode_jac_sparse = semidiscretize(semi_float_type, tspan,
+                                jac_prototype = jac_prototype,
+                                colorvec = coloring_vec)
+
+###############################################################################
+### callbacks & solve ###
+
+summary_callback = SummaryCallback()
+analysis_callback = AnalysisCallback(semi_float_type, interval = 50)
+alive_callback = AliveCallback(alive_interval = 3)
+
+# Note: No `stepsize_callback` due to implicit solver
+callbacks = CallbackSet(summary_callback, analysis_callback, alive_callback)
+
+sol = solve(ode_jac_sparse, # using `ode_jac_type` is essentially infeasible, even single step takes ages!
```

Comment:
Can you even run the example with `ode_jac_type`? It only supports the tracer type and should obviously not be used to get numerical results?